### PR TITLE
Fix typescript error for the error messages in AssetConnection

### LINKED ISSUE #151: [AAT JS] Typescript errors on AssetConnection.ts
I am getting the following errors when I run `npm run build` at the root of the project.
image

### PR DESCRIPTION
**Pull request by stmoreau**
Somewhat of a lazy approach to fix some typescript type errors for the reason variable in catch blocks.

Resolves https://github.com/ably/ably-asset-tracking-js/issues/151.

### CODE REVIEW

File: src/lib/AssetConnection.ts [hunk 1]
```diff
@@ -77,8 +77,8 @@ class AssetConnection {
         resolution: this.resolution,
       })
       .catch((reason) => {
-        this.logger.logError(`Error entering channel presence: ${reason}`);
-        throw new Error(reason);
+        this.logger.logError(`Error entering channel presence: ${JSON.stringify(reason)}`);
```
Comment (owenpearson) [inline src/lib/AssetConnection.ts:80]: I actually think `reason` or `reason.message` would be better here. The reason will be an ably-js errorinfo which has a `toString` convenience method which will look a bit nicer than the JSON string. I think this is probably the best fix anyway:
```suggestion
        this.logger.logError(`Error entering channel presence: ${(e as AblyTypes.ErrorInfo).message}`);
```

**Review by owenpearson — CHANGES_REQUESTED** [review]
Looks good! just one thing i would change:

File: src/lib/AssetConnection.ts (hunk 1 above)
Comment (stmoreau) [reply src/lib/AssetConnection.ts]: Oh nice one! Updated 😄

Comment (owenpearson) [reply src/lib/AssetConnection.ts]: sorry forgot to mention - the same applies to the other JSON.stringify you added 😄

Comment (stmoreau) [reply src/lib/AssetConnection.ts]: of course 🤦 done!

**Review by owenpearson — APPROVED** [review]
LGTM, nice one!

**Pull request merged**